pkg/healthcheck/healthcheck.go

```diff
@@ -499,3 +516,46 @@ func validateDataPlanePods(pods []v1.Pod, targetNamespace string) error {
 
 	return nil
 }
+
+func validateDataPlanePodReporting(k8sPods []v1.Pod, promPods []*pb.Pod, targetNamespace string) error {
+	k8sMap := map[string]struct{}{}
+	promMap := map[string]struct{}{}
+
+	for _, p := range k8sPods {
+		k8sMap[p.Namespace+"/"+p.Name] = struct{}{}
+	}
+	for _, p := range promPods {
+		// the `Added` field indicates the pod was found in Prometheus
+		if p.Added {
+			promMap[p.Name] = struct{}{}
+		}
+	}
+
+	onlyInK8s := []string{}
+	for k := range k8sMap {
+		if _, ok := promMap[k]; !ok {
+			onlyInK8s = append(onlyInK8s, k)
+		}
+	}
+
+	onlyInProm := []string{}
+	for k := range promMap {
+		if _, ok := k8sMap[k]; !ok {
+			onlyInProm = append(onlyInProm, k)
+		}
+	}
+
+	errMsg := ""
+	if len(onlyInK8s) > 0 {
+		errMsg = fmt.Sprintf("Data plane metrics not found for %s. ", strings.Join(onlyInK8s, ", "))
+	}
+	if len(onlyInProm) > 0 {
+		errMsg += fmt.Sprintf("Found data plane metrics for %s, but not found in Kubernetes.", strings.Join(onlyInProm, ", "))
```

Comment:
Innnnteresting. I'm not sure this error will ever fire, since the ListPods endpoint builds its list of pods from the Kubernetes API, so if a host is not found in the Kubernetes, it also won't exist in the response from the ListPods endpoint. But it certainly doesn't hurt to leave this in here just in case something is really wonky.

Reply:
I was thinking maybe we could have used the prometheus endpoint `api/v1/targets`'s results and compare that with the data plane pods to verify that data plane pods are in fact reporting stats. But, considering that we would have to deal with unmarshalling JSON and other things to make the endpoint's result consumable, this is probably a better way to do this.

Reply:
@dadjeibaah Yup I definitely thought about that approach. I think if ListPods at some point changes in behavior, we could revisit which Prometheus endpoint we use.